### ISSUE TITLE
Publicize Custom Message Doesn't Show In Twitter

### ISSUE THREAD

**Issue by gwensmithx** [Automattic/jetpack#11752]
I didn't see this issue in the related issues so I decided to post. I hope I'm posting in the correct spot! I was trying to replicate an issue a customer had with displaying the Custom Message in Twitter from the new Publicize feature in Gutenberg. The custom message was not showing, but the rest of the post was posted to Twitter successfully.

#### Steps to reproduce
1. Starting at URL: https://gwentestsite733700880.wordpress.com/2018/12/09/test-post-title/
^ This is my test site. I installed the theme TheFour and used my Premium Plan.
2. Create a new blog post using Gutenberg editor.
3. Add in featured image and content. Clicked JP Icon to publicize and connected my Twitter account (@Nvision_WebDev) and put in my custom message.
4. Clicked "Publish" post - looked everything over (JP was connected - message was in), confirmed Publishing.
![screenshot-gwentestsite733700880 wordpress com-2018 12 09-15-11-47](https://user-images.githubusercontent.com/5779512/49703851-9ca62d00-fbc7-11e8-9399-8feb53c5c2c5.png)

5. Checked my twitter account to see post. Post was posted with my featured image and snippet of content but no custom message.

![screenshot-twitter com-2018 12 09-15-32-49](https://user-images.githubusercontent.com/5779512/49703873-daa35100-fbc7-11e8-81d4-071fceeadb69.png)

What I expected:
I expected to see the Custom Message from Publicize in my Twitter Post.

What happened instead:
The post was posted without the Custom Message.

Browser / OS version:
Mozilla Firefox Quantum - Windows 10

## Temporary Workaround

Steps shared ([h/t](https://github.com/Automattic/jetpack/issues/11752#issuecomment-540618420): @marutim):

1. Write the post.
2. Write the custom message for Twitter.
3. Hit the "Save draft" and wait for a confirmation message that the draft has been saved.
4. Hit the "Publish" button only after seeing confirmation that the draft is saved.

**Comment by jeherve** [comment]
I can't seem to reproduce on my end right now. If I add a custom message to the Publicize custom message input field, that custom message is used on Twitter. Here is an example with a Jetpack site:

https://twitter.com/jeherve/status/1110503318672101376

And here is an example with a WordPress.com simple site:

<img width="616" alt="screenshot 2019-03-29 at 14 33 19" src="https://user-images.githubusercontent.com/426388/55236206-b7acb380-522f-11e9-9549-96f686c3f6b2.png">

In both cases, I was using the Block editor.

Noting that I tested with Twitter, as per the issue's title. I did not test with LinkedIn. I am mentioning this since one of the screenshots above was a LinkedIn profile page.

@gwensmithx Can you still reproduce the issue? If so, could you make a screenshot of your custom Publicize message before you publish your post?

Thank you!

**Comment by jruud** [comment]
Another report of this in #1959253-zen

I asked for specifics on the posts that were affected, because the behavior was also inconsistent on the user's site.

**Comment by susanjanec** [comment]
Another report of a user encountering this where instead of the excerpt text appearing, it was just the title of the post with the link to the post.

The instance where the user published post with excerpt text and it shows just the title of the post and link to the post - https://cld.wthms.co/OSb7Tp

Excerpt text from post here - https://cld.wthms.co/8cPIxp 

Follow up ticket 1986944-zen

**Comment by jeherve** [comment]
@susanjanec Do you happen to know which editor the author used to publish their post?

**Comment by susanjanec** [comment]
@jeherve They were using the Block editor.

**Comment by jeherve** [comment]
Sorry, I should have been clearer with my question. Was this the editor in Calypso or the one in wp-admin?

**Comment by susanjanec** [comment]
No worries @jeherve. It was the editor in Calypso.

**Comment by jeherve** [comment]
I am afraid I still can't reproduce, whether I am on a Jetpack site or a WordPress.com site, and when using the Calypso editor, the Block Editor in Calypso, or the block editor in wp-admin. If I specify a custom message in the Publicize settings, it is used by Facebook and Twitter instead of the post title.

I must consequently be doing something different from the post authors above. It would be good to know if:
- They post right away or schedule.
- They saved their drafts and came back to it later, or open the editor, wrote and published right away.
- At what point in the process they entered the custom Publicize message.

Basically, having a detailed series of steps I could reproduce would be super helpful!

**Comment by susanjanec** [comment]
Hi @jeherve they mentioned that they copy and paste the text they want to show on Twitter and Facebook in the excerpt text box before they publish the post.

They didn't mention if the post was published right away or it was scheduled. The user never replied back at this point, so it may not be an issue anymore?

**Comment by jeherve** [comment]
> The user never replied back at this point, so it may not be an issue anymore?

That's possible. Let me know if you ever get a reply with more info!

Thanks!

**Comment by jenlynnemc** [comment]
@jeherve User followed up to the questions above on how they post:
I’ve tried all options you listed in the first two bullets 
(- They post right away or schedule.
- They saved their drafts and came back to it later, or open the editor, wrote and published right away.)

over time but in this latest instance I posted right away and I added my custom publicise message at the point I was ready to publish the post and just before hitting publish.

Hope this helps. The user is also willing to do more testing or screenshare if needed:
https://woothemes.zendesk.com/agent/tickets/2127118

**Comment by jeherve** [comment]
@jenlynnemc Thank you!

To try to summarize all this:

- Conversation happened in 1630181-zen and then 2127118-zen
- Blog ID 13900285 (wpcom simple)
- Publicize connected to LinkedIn and Twitter
- They added the message right before publishing
- They can reproduce (not a one-time thing)
- Gutenberg is enabled for that site, i.e. when they load wordpress.com/post/ they'll be redirected to wordpress.com/block-editor/post for that site (Gutenberg in Calypso).
- Looking at the email exchanges, we have not asked them where they write their posts, though. They have 2 main options:
    - in wp-admin (`https://site.wordpress.com/wp-admin/post-new.php`)
    - in Calypso (`https://wordpress.com/block-editor/post/site.wordpress.com`)

Could you ask them about this? Could you also ask them to try adding the custom message, then hitting "Save draft", and only once they get the confirmation that the draft is saved, hit the Publish button.

This should allow us to narrow things down and hopefully reproduce the problem.

Thank you!

**Comment by SiobhyB** [comment]
I've sent those questions to the user in 2127118-zen. :)

**Comment by kriskorn** [comment]
@jeherve These steps helped user get everything working:

1. Open Calypso: https://wordpress.com/block-editor/post/innerventur.es
2. Add the custom message.
3. Hit the "Save draft" and wait for a confirmation message that the draft has been saved.
4. Hit the "Publish" button only after seeing confirmation that the draft is saved.

However, now user has another concern/issue and I am not sure if this is the expected behaviour. 

They have published this post - https://innerventur.es/2019/08/09/homo-deus-and-the-cosmic-dance-between-humans-mind-and-machine/

When they now share it in social media, they get these results:

![sharing-linkedin-and-twitter](https://user-images.githubusercontent.com/12064241/63090211-3eee8f80-bf63-11e9-87e0-3ef9d88a5158.png)

Linkedin on the left and Twitter on the right. The custom message is now replacing the post title when shared. Shouldn't this be added as an addition when sharing? 

**Post Title:** HOMO DEUS AND THE COSMIC DANCE BETWEEN HUMANS, MIND AND MACHINE

**Social media title:** After reading Homo Deus on vacation, a #thoughtrocket and #daneldoodle on the cosmic dance between humans, mind and machine

Twitter shows it a bit better, but Linkedin just repeats it two times and removes the Title altogether.

**Comment by jeherve** [comment]
@kriskorn This is indeed the intended behaviour, but I think it may be nice to reconsider it for LinkedIn. I agree that it may be best to have the post title displayed below the image, so as to avoid duplicates. I don't know if that's something in our control since the post preview is created by LinkedIn, but it would be something worth looking into. I would recommend that you create a new issue to keep track of this.

**Comment by jessestu** [comment]
This bug came up in 2394642-zen (that's the followup email for our 8441736-hc chat).

__TL;DR__—I thought this was plan related, but I'm probably wrong.

They have the Personal plan. All of their posts so far have been affected by this bug. I temporarily added the Premium plan for them and had them test this through the re-Publicize feature, and it worked as expected. I then removed the plan and we tried again from the editor on a new test post, and it posted only the title.

That led me to think it was plan-related, but now I see that's probably because we re-Publicized.

_Next:_ I've asked the user to test saving the draft and then publishing and letting us know what happens.

**Comment by marutim** [comment]
Another report of this issue here: ZD-2394642

The following steps helped the user:
1. Write the post.
2. Write the custom message for Twitter.
3. Hit the "Save draft" and wait for a confirmation message that the draft has been saved.
4. Hit the "Publish" button only after seeing confirmation that the draft is saved.

**Comment by jessestu** [comment]
@jeherve Thanks for the reply! I'll let @marutim share any updates they experienced. They were replying to the user reporting success from [my comment above](https://github.com/Automattic/jetpack/issues/11752#issuecomment-537696808). The user was successful with the workaround to save the draft before publishing. Here was my experience of your questions when working with the customer:

> Are you able to reproduce consistently?

I was not able to repro on my own site. I tested it on a WPcom Premium plan.

> Is that site running the most recent version of Jetpack?

The user has WPcom Personal plan, so I'm assuming it's not applicable.

> Does it still happen when all other plugins are deactivated?

N/A given WPcom simple site.

Thank you for your work on this!

**Comment by wpericam** [comment]
Hashtags and @mentions not showing when post is shared via publicize via Premium .com site. 16640602-hc

**Comment by thabotswana** [comment]
Another report of this here: #17952949-hc, WP.com Premium plan

**Comment by formosattic** [comment]
It came up in 2620263-zen, WPCOM Premium plan.

I was also able to replicate on a WPCOM Free plan, but Marutim's workaround worked for me.

Noting that this appears to be a duplicate or related to #14097? (cc @jeherve for triaging or another look at the issue)

**Comment by mikeicode** [comment]
Also reported 15640131-hc Premium plan, I am sending the above workaround now.

**Comment by donalirl** [comment]
Another case of this in #2690023-zen. What we know:

* They add a custom message for the tweet before publishing the post for the first time
* Yet, only the post title, link, and image are tweeted out.
* They use block editor on a WPcom simple site
* They delete the tweet, switch the post back to draft, disconnect and re-connect twitter, then publish the post again which works to show the custom message
* Happens repeatedly for every new post they publish

Asking them to try the workaround mentioned above and get back to us.

**Comment by supernovia** [comment]
Noting I've reproduced this on my own .com site. Custom social media was set but not included on Twitter. The title tweeted out instead.  I'll test again on my own site with the workaround listed above to see what happens.

I was troubleshooting with this user:
https://en.forums.wordpress.com/topic/jetpack-custom-message-does-not-appear-in-twitter/#post-3420054

**Comment by amgmbs** [comment]
> @marutim Are you able to reproduce consistently? Is that site running the most recent version of Jetpack? Does it still happen when all other plugins are deactivated?

@jeherve I have a feeling that this does not work in the latest versions of jetpack

https://gist.github.com/jeherve/7990726

**Comment by jeherve** [comment]
> I have a feeling that this does not work in the latest versions of jetpack

I have not tested it, but if it fails it may be because of the issue outlined in https://github.com/Automattic/jetpack/issues/14269#issuecomment-580004663

**Comment by pauljacobson** [comment]
I have a similar report in 2720674-zen where the user reported the following:

>Every time I try to share a new post on Facebook it ignores the custom message and publish the whole post on my FB page. On twitter it also ignores the custom message, tweeting the post's title instead

I've asked the user to reset their Twitter and Facebook connections, and then take screenshots of the custom message, and resulting posts on social media so we can investigate further.

**Comment by hacchism** [comment]
Another case reported in 18752548-hc. And I could reproduce it on my test simple sites. I shared the workaround @marutim shared [here](https://github.com/Automattic/jetpack/issues/11752#issuecomment-540618420) for now.

**Comment by rickmgithub** [comment]
2720674-zen (update).

Copied from - https://github.com/Automattic/jetpack/issues/11752#issuecomment-586913063
> Every time I try to share a new post on Facebook it ignores the custom message and publish the whole post on my FB page. On twitter it also ignores the custom message, tweeting the post's title instead
> 
> I've asked the user to reset their Twitter and Facebook connections, and then take screenshots of the custom message, and resulting posts on social media so we can investigate further.
> 

The user has provided some screenshots in their ticket to show what happens.

I've also shared @marutim suggestion

**Comment by JoshuaGoode** [comment]
Another instance reported today. Provided [workaround](https://github.com/Automattic/jetpack/issues/11752#issuecomment-540618420).


internal: 18976687-hc

**Comment by tijosh** [comment]
Another instance in #2754421-zen

Shared [workaround](https://github.com/Automattic/jetpack/issues/11752#issuecomment-540618420)

Block editor, Twenty Fourteen theme

**Comment by rinazrina** [comment]
Another report: 19114957-hc
Follow up: 2765834-zen

The [workaround](https://github.com/Automattic/jetpack/issues/11752#issuecomment-540618420) worked on my test site (simple site). Shared it to the user too.

**Comment by pranali333** [comment]
Similar report here: 2780932-zen

- Using the classic interface to publish the new post.
- Workaround for saving the post as a draft before publishing helped.

The user wants to know, how to fix this going further?

**Comment by sophiegyo** [comment]
We got another report in 2827362-zen

* Block editor, both on mobile and computer
* Suggested the save-as-draft workaround, and mentioned the share option in Calypso.

**Comment by sophiegyo** [comment]
I think - am not sure - that we got another report in 20089073-hc.

I shared the save-as-draft workaround.

**Comment by chaitanyamsv** [comment]
Another report in 3001543-zen

**Comment by nagpai** [comment]
Maybe related, for WP.com simple sites
https://github.com/Automattic/wp-calypso/issues/38697

[Workaround suggested](https://github.com/Automattic/jetpack/issues/11752#issuecomment-540618420)  helped.

**Comment by naxoc** [comment]
Another report in 3140379-zen

**Comment by zdenys** [comment]
Another case in 3156201-zen. Provided [the workaround above](https://github.com/Automattic/jetpack/issues/11752#issuecomment-540618420).

**Comment by zdenys** [comment]
Another case 3187472-zen. Provided [the workaround above](https://github.com/Automattic/jetpack/issues/11752#issuecomment-540618420).

**Comment by cathymcbride** [comment]
related or possibly duplicate issues:  
38697-gh-jetpack
https://github.com/Automattic/jetpack/issues/14269
https://github.com/Automattic/wp-calypso/issues/34002
45114-core <-- likely the root issue

**Comment by wynwin** [comment]
Another one 3310080-zen

**Comment by smithkyle45** [comment]
User reported this in 3329751-zen, provided a screen recording detailing how the custom message was not used when creating a new post. 

Shared the workaround mentioned in this thread.

**Comment by supernovia** [comment]
I found another case here: https://github.com/Automattic/jetpack/issues/11752

**Comment by cathymcbride** [comment]
@supernovia 11752 is this issue? Copy/paste error?

**Comment by dani-elo** [comment]
I found another one here 3390585-zen. [Provided the workaround above.](https://github.com/Automattic/jetpack/issues/11752#issuecomment-540618420)

**Comment by jerrysarcastic** [comment]
Also reported by a site owner in the [WordPress.com forums](p24uxq-2M-p2)

**Comment by Joennuh** [comment]
I also experience this issue from time to time. It just uses the title of the post instead of my custom message. I will try the workaround to first save a draft with the custom message, waiting for confirmation and then publish the post.
I'm not that experienced with Github but i see some <number>-zen's here. Do I need to do something special to report it or is this message enough?

**Comment by jeherve** [comment]
> Do I need to do something special to report it or is this message enough?

This is enough, thank you!

**Comment by Joennuh** [comment]
> 
> 
> > Do I need to do something special to report it or is this message enough?
> 
> This is enough, thank you! You'll get notifications as soon as we start working on this.

Thanks! I hope it will be fixed soon. It would save some time for performing the additional steps.

**Comment by carladoria** [comment]
anther report here [13852883-hc](13852883-hc)

**Comment by ezebecke** [comment]
Also, [23579779-hc](https://mc.a8c.com/support-stats/happychat/transcript.php?id=23579779)

**Comment by syedshaon** [comment]
Another issue 3480592-hc

**Comment by jeherve** [comment]
Noting that https://core.trac.wordpress.org/changeset/49172 may help us fix this, at least for our users who use WordPress 5.6. Maybe we can update Sync accordingly.

Related: #14269

**Comment by jeherve** [comment]
This should be fixed in Jetpack 9.2 for folks using WordPress 5.6 (scheduled to be released on December 8).
